docs/quickstarts/nextjs.mdx

```diff
@@ -277,83 +268,113 @@ export default function Example() {
+      return NextResponse.json({ userId }, { status: 200 });
+    }
+    ```
+  </Tab>
+  <Tab>
+    You can protect your API routes by using the [`auth()`](/docs/references/nextjs/auth) helper, if the route is not protected by [middleware](/docs/references/nextjs/auth-middleware). 
 
-  // use the user object to decide what data to return
+    In some cases, you may need the full [`User`][user-object] object. For example, if you want to access the user's email address address or name, you can use  [`currentUser()`](/docs/references/nextjs/current-user) helper to get the full [`User`][user-object] object. Under the hood, this calls `fetch()` so it is automatically deduped per request.
 
-  return res.status(200).json({});
-}
-```
+   ```tsx filename="app/api/user/route.ts"
+    import { NextResponse } from 'next/server';
+    import { currentUser, auth } from "@clerk/nextjs";
 
-### Server side render user data
+    export async function GET(req: Request, res: NextResponse) {
+      const { userId } = auth();
 
-<Tabs items={["App Router", "Pages Router"]}>
+      if (!userId) {
+        return new NextResponse("Unauthorized", { status: 401 });
+      }
+
+      const user = await currentUser();
+
+      // Perform your Route Handler's logic with the returned user object
+
+      return NextResponse.json({ "user": user }, { status: 200 })
+    }
+  ```
+  </Tab>
+</Tabs>
+
+#### Pages Router
+
+<Tabs items={["API Route", "API Route w/ User Fetch", "getServerSideProps"]}>
   <Tab>
-    #### **`currentUser()`**
+    For Next.js apps using the Pages Router, you can protect your API routes by using the [`getAuth`](/docs/references/nextjs/get-auth) helper and retrieve data from your own systems.
 
-    [`currentUser()`](/docs/references/nextjs/current-user) loads the `User` object for the authenticated user from Clerk's backend API. This is helpful if you want to render information, like their first and last name, directly from the server.
+    ```tsx filename="pages/api/auth.ts"
+    import type { NextApiRequest, NextApiResponse } from "next";
+    import { getAuth } from "@clerk/nextjs/server";
 
-    Under the hood, this calls `fetch()` so it is automatically deduped per request.
+    export default function handler(req: NextApiRequest, res: NextApiResponse) {
+      const { userId } = getAuth(req);
 
-    ```tsx filename="page.tsx"
-    import { currentUser } from "@clerk/nextjs";
-    import type { User } from "@clerk/nextjs/api";
+      if (!userId) {
+        return res.status(401).json({ error: "Unauthorized" });
+      }
 
-    export default async function Page() {
-      const user: User | null = await currentUser();
-      // ...
+      // retrieve data from your database
+  
+      res.status(200).json({});
     }
     ```
+  </Tab>
+  <Tab>
+    For Next.js apps using the Pages Router, you can protect your API routes by using the [`getAuth`](/docs/references/nextjs/get-auth) helper and retrieve data from your own systems.

```

Comment:
What do we mean by "and retrieve data from your own systems"?

Reply:
That was from the previous version. I'll revise.

Reply:
Reminder to revise this wording 👀